lib/inspec/resources/podman.rb

```diff
@@ -0,0 +1,337 @@
+
+    def info
+      return @info if defined?(@info)
+
+      sub_cmd = "info"
+      output = run_command(sub_cmd)
+      @info = Hashie::Mash.new(JSON.parse(output))
+    rescue JSON::ParserError => _e
+      Hashie::Mash.new({})
+    end
+
+    # returns information about podman objects
+    def object(id)
+      return @inspect if defined?(@inspect)
+
+      output = run_command("inspect #{id}")
+      data = JSON.parse(output)
+      data = data[0] if data.is_a?(Array)
+      @inspect = Hashie::Mash.new(data)
+    rescue JSON::ParserError => _e
+      Hashie::Mash.new({})
+    end
+
+    def to_s
+      "Podman"
+    end
+
+    private
+
+    # Calls the run_command method to get all podman containers and parse the command output.
+    # Returns the parsed command output.
+    def get_containers
+      sub_cmd = "ps -a --no-trunc"
+      output = run_command(sub_cmd)
+      parse(output)
+    end
+
+    # Calls the run_command method to get all podman images and parse the command output.
+    # Returns the parsed command output.
+    def get_images
+      sub_cmd = "images -a --no-trunc"
+      output = run_command(sub_cmd)
+      parse(output)
+    end
+
+    # Calls the run_command method to get all podman network list and parse the command output.
+    # Returns the parsed command output.
+    def get_networks
+      sub_cmd = "network ls --no-trunc"
+      output = run_command(sub_cmd)
+      parse(output)
+    end
+
+    # Calls the run_command method to get all podman pod list and parse the command output.
+    # Returns the parsed command output.
+    def get_pods
+      sub_cmd = "pod ps --no-trunc"
+      output = run_command(sub_cmd)
+      parse(output)
+    end
+
+    # Calls the run_command method to get all podman volum list and parse the command output.
+    # Returns the parsed command output.
+    def get_volumes
+      sub_cmd = "volume ls"
+      output = run_command(sub_cmd)
+      parse(output)
+    end
+
+    # Runs the given podman command on the host machine on which podman is installed
+    # Returns the command output or raises the command execution error.
+    def run_command(subcommand)
+      result = inspec.command("podman #{subcommand} --format json")
+      if result.stderr.empty?
+        result.stdout
+      else
+        raise "Error while running command \'podman #{subcommand}\' : #{result.stderr}"
+      end
+    end
+
+    # Method to parse JDON content.
+    # Returns: Parsed data.
+    def parse(content)
+      require "json" unless defined?(JSON)
+      output = JSON.parse(content)
+      parsed_output = []
+      output.each do |entry|
+        entry = entry.map do |k, v|
+          [k.downcase, v]
+        end.to_h
+        parsed_output << entry
+      end
+      parsed_output
+    rescue => e
+      raise Inspec::Exceptions::ResourceFailed, "Unable to parse command JSON output: #{e.message}"
+    end
+  end
+
+  # class for podman.containers plural resource
+  class PodmanContainerFilter
```

Comment:
Hey can we separate these classes to separate files and require them maybe in this class if it is needed here?

Reply:
We require them here to work with podman.container syntax.

Reply:
They are part of podman resource.